loop_refine.hpp integration patch.

Guards ICP and records results.

--- src/loop_refine.hpp
+++ src/loop_refine.hpp
@@ -4,6 +4,8 @@
 #include "tools.hpp"
 #include "voxel_map.hpp"
 
+#include <voxelslam/offline_bridge.hpp>
+
 // #include "STDesc.h"
 #include <gtsam/geometry/Pose3.h>
 #include <gtsam/slam/PriorFactor.h>
@@ -54,6 +56,9 @@
     pi.x = ap.x; pi.y = ap.y; pi.z = ap.z;
     input_cloud.push_back(pi);
   }
+  if (pl_src.empty() || input_cloud.empty()) {
+    return false;
+  }
   kd_tree.setInputCloud(input_cloud.makeShared());
 
   vector<int> pointIdxNKNSearch(1);
@@ -138,8 +143,10 @@
 
   Eigen::SelfAdjointEigenSolver<Eigen::Matrix3d> saes(mat_norm);
   Eigen::Vector3d eig_vec = saes.eigenvalues();
+  bool passed = eig_vec[0] > icp_eigval && is_converge == 1;
+  voxelslam_offline::record_loop_icp_result(eig_vec[0], eig_vec[1], eig_vec[2], is_converge, passed, match_num);
   printf("eigvalue: %lf %lf %lf %d\n", eig_vec[0], eig_vec[1], eig_vec[2], is_converge);
-  return eig_vec[0] > icp_eigval && is_converge == 1;
+  return passed;
   // return eig_vec[0] > icp_eigval;
 
 }
